### ISSUE TITLE
ReduceLROnPlateau Not Compatible with torch.save

### ISSUE THREAD

**Issue by theforager** [pytorch/pytorch#3891]
I'm trying to save the various objects necessary to resume training as I'm using EC2 spot instances, and one component of this is the learning rate scheduler. While `torch.save` works fine with `StepLR` and other basic schedulers, it doesn't not appear to work with `ReduceLROnPlateau` due to the lambda function `_init_is_better`, which `pickle` can't handle.

From the source, it appears this a hidden error catching function that isn't used anywhere in the code base. Given that it's causing incompatibility with other PyTorch functions and other `lr_scheduler`, it seems like it would be best to either:
 (1) remove this function and class property or
 (2) fully implement this functionality in a pickle compatible way.

I'd be happy to make the change and submit a pull request but wanted to make sure I'm not missing something here.
